docs/spikes/centops-adapter-spike.md

```diff
@@ -0,0 +1,77 @@
+# Introduction
+
+This document outlines what we learnt in the spike [DMR: CentOps-to-NGINX converter [Spike]](https://github.com/buerokratt/DMR/issues/40)
+
+## Background
+
+DMR needs to keep its participant routing table up to date.
+The single source of truth for participant information is CentOps, which makes this information available via a REST endpoint.
+Consuming a REST endpoint and updating the routing table is not a problem for the .NET based DMR, but it is still unclear what would be the preferred way to do
+this in Nginx based DMR where the participant routing table is stored as a configuration file.
+
+## Spike goals
+
+Figure out what are the possible solutions for updating DMR-Nginx participant routing table.
+
+## Current situation
+
+Participant routing is achieved via an upstream block in nginx configuration, e.g.
+
+```
+upstream Classifier {
+    server 127.0.0.1:7140;
+}
+
+upstream MockBot {
+    server 127.0.0.1:7285;
+}
+```
+
+Updating this information requires modifying and reloading the nginx configuration file (`nginx -s reload`).
+
+## Options
+<style>
+r { color: Red }
+g { color: Green }
+</style>
+
+### Nginx Plus API
+Nginx Plus has an API for modifying upstream configuration on-the-fly without reloading configuration or processes -
```

Comment:
We'll stick to only using free software as our core components. But having this information brought out here is OK.